src/IECoreArnold/Renderer.cpp

```diff
@@ -1788,11 +1790,13 @@ class ArnoldAttributes : public IECoreScenePreview::Renderer::AttributesInterfac
 				case IECoreScene::MeshPrimitiveTypeId :
 					m_polyMesh.hash( meshInterpolationIsLinear, h );
 					m_displacement.hash( h );
+					h.append( m_automaticInstancing );
```

Comment:
Just checking I understand why this is included in the hash here : so that in `ArnoldAttributes::apply()` we will reject attribute edits that change the instancing status, and therefore will require that the geometry is re-sent? If that's the case, wouldn't it apply to all object types - from memory, we do auto-instance all types I think?

Reply:
I have now applied it to all types, that does make more sense, and I've confirmed that it now works on external procedurals.  The hash was in fact not working before ... I would have expected it to work on procedurals at least because I had put it in every category I was aware of needing it, including the case for IECoreScene::ExternalProceduralTypeId ... I was surprised to find out that in ArnoldAttributes::apply, we assign an id of IECore::InvalidTypeId to external procedurals, and reserve ExternalProceduralTypeId for volumes.  Now that I've moved my flag though, everything is working fine, so maybe how these types work is intentional?